fe/fe-core/src/main/java/org/apache/doris/nereids/util/JoinUtils.java

```diff
@@ -484,4 +485,23 @@ public static boolean checkReorderPrecondition(LogicalJoin<?, ?> top, LogicalJoi
         markSlots.retainAll(bottom.getOutputSet());
         return markSlots.isEmpty();
     }
+
+    /**
+     * Use the children nullable property of the join to adjust the slot used by the join conjuncts.
+     * Such as 'a left join b left join c where b.condition > 1'
+     * if the join change to '(b left join c) right a where b.condition > 1',
+     * the nullable property of b.condition should be false
+     */
+    public static LogicalJoin<Plan, Plan> adjustJoinConjunctsNullable(LogicalJoin<Plan, Plan> join) {
+        Map<Slot, Slot> childSlotMap = new HashMap<>();
+        for (Plan child : join.children()) {
+            for (Slot slot : child.getOutput()) {
+                childSlotMap.put(slot, slot);
```

Comment:
should not replace directly, generate a new slot with correct nullable is better, could reuse or refer to `fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/AdjustNullable.java`